### PR TITLE
Fixes VSTS Bug 1000977: Make preference dialog more resilient to

### PR DESCRIPTION
**Pull request by mkrueger**
extension errors

https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1000977

### CODE REVIEW

File: main/src/core/MonoDevelop.Ide/ExtensionModel/ItemOptionPanels.addin.xml [hunk 1]
```diff
@@ -26,6 +26,7 @@
 			<Condition id="ItemType" value="Project">
 				<Panel id = "GeneralProjectPanel" _label = "Project" class = "MonoDevelop.Ide.Projects.OptionPanels.GeneralProjectOptions" />
 				<Panel id = "BaseDirectoryPanel" _label = "Location of Project Files" class = "MonoDevelop.Ide.Projects.OptionPanels.BaseDirectoryPanel"/>
+				<Panel id = "fooPanel" _label = "Foo" class = "MonoDevelop.Ide.Projects.OptionPanels.FooPanel" />
```
Comment (mrward) [inline main/src/core/MonoDevelop.Ide/ExtensionModel/ItemOptionPanels.addin.xml]: We should remove this. I guess this was used for testing.

Comment (mkrueger) [reply main/src/core/MonoDevelop.Ide/ExtensionModel/ItemOptionPanels.addin.xml]: argh y.

File: main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Gui.Dialogs/OptionsDialog.cs [hunk 2]
```diff
@@ -656,7 +656,13 @@ SectionPage CreatePage (TreeIter it, OptionsDialogSection section, object dataOb
...
 				if (pi == null) {
-					IOptionsPanel panel = node.CreatePanel ();
+					IOptionsPanel panel;
+					try {
+						panel = node.CreatePanel ();
+					} catch (Exception e) {
+						LoggingService.LogInternalError ("Error while creating options panel: " + node.Id, e);
+						continue;
+					}
 					pi = new PanelInstance ();
```
Comment (sandyarmstrong) [inline main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Gui.Dialogs/OptionsDialog.cs]: Do you need a `panel == null` check here, like below with the widget? Or is it ok to have a null `panel`?

Comment (mkrueger) [reply main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Gui.Dialogs/OptionsDialog.cs]: added a check - y can happen during options panel development.

**Pull request merged**